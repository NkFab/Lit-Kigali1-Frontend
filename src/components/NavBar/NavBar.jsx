import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from '../common/Button/Button';
import logoWhite from '../../assets/images/logo-white.png';
import userAvatar from '../../assets/images/avatar.png';

export class NavBar extends Component {
  state = {
    showMenu: false,
  };

  showMenu = () => {
    const { showMenu } = this.state;
    this.setState(
      {
        showMenu: !showMenu,
      },
      () => {
        document.addEventListener('click', this.closeMenu);
      },
    );
  };

  closeMenu = () => {
    this.setState({ showMenu: false }, () => {
      document.removeEventListener('click', this.closeMenu);
    });
  };

  renderLinks = parentClass => (
    <ul className={parentClass}>
      <li>
        <Link to="/articles/create">New article</Link>
      </li>
      <li>
        <Link to="/articles">Articles</Link>
      </li>
      <li>
        <Link to="/my-articles">My Articles</Link>
      </li>
      <li className="separator" />
      <li>
        <Link to="/settings">Settings</Link>
      </li>
      <li>
        <Link to="/dashboard">Dashboard</Link>
      </li>
      <li>
        <Button classes="transparent signout-btn">Sign out</Button>
      </li>
    </ul>
  );

  render() {
    const { showMenu } = this.state;
    const { isLoggedIn } = this.props;
    return (
      <nav className="top-navbar container">
        <div className="col-3-mob">
          <Link to="/">
            <img src={logoWhite} alt="brand-logo" className="brand-logo" />
          </Link>
        </div>
        <div className="col-9-mob content-right color-white">
          <div className="toggle-menu is-tablet">
            <button className={`hamburger ${showMenu ? 'active' : ''}`} onClick={this.showMenu} />
            <div className={`hamburger-nav ${showMenu ? 'active' : ''}`}>{this.renderLinks()}</div>
          </div>
          <div className="nav-button navbar-dropdown is-desktop color-white">
            <Button onClick={this.showMenu} classes="transparent">
              {isLoggedIn ? (
                <img src={userAvatar} className="top-navbar__avatar" alt="User logo" />
              ) : (
                ''
              )}
            </Button>
            {this.renderLinks('dropdown')}
          </div>
        </div>
      </nav>
    );
  }
}

NavBar.protoTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
};

export const mapStateToProps = ({ currentUser: { isLoggedIn } }) => ({
  isLoggedIn,
});

export default connect(mapStateToProps)(NavBar);
