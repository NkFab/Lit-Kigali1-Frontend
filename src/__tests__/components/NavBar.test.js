import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapStateToProps } from '../../components/NavBar/NavBar';

describe('<NavBar />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });

  test('should render the NavBar', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the NavBar', () => {
    expect(wrapper.state().showMenu).toBeFalsy();
  });

  describe('when clicking on the `hamburger button`', () => {
    beforeEach(() => {
      wrapper.find('.hamburger').simulate('click');
    });
    afterEach(() => {
      wrapper.setState({ showMenu: false });
    });

    test('changes the `showMenu state` to true', () => {
      expect(wrapper.state().showMenu).toBeTruthy();
    });

    test('changes the `showMenu state` to false', () => {
      wrapper.find('.hamburger').simulate('click');
      expect(wrapper.state().showMenu).toBeFalsy();
    });

    test('changes the `showMenu state` to false by calling `showMenu` function', () => {
      wrapper.instance().showMenu();
      expect(wrapper.state().showMenu).toBeFalsy();
    });

    test('changes the `showMenu state` to false by calling `closeMenu` function', () => {
      wrapper.instance().closeMenu();
      expect(wrapper.state().closeMenu).toBeFalsy();
    });
  });

  describe('reducers', () => {
    test('should return `mapStateToProps`', () => {
      const initialState = {
        currentUser: {
          isLoggedIn: false,
        },
      };
      const state = mapStateToProps(initialState);
      expect(state).toEqual(initialState.currentUser);
    });
  });
});
