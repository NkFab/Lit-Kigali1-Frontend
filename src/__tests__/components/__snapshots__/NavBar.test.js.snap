// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render the NavBar 1`] = `ShallowWrapper {}`;
